src/App.tsx

```diff
@@ -1,24 +1,124 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { USER_ID } from './constants';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorType } from './types/Error';
+import { Filter } from './types/Filter';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<ErrorType>(ErrorType.NONE);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [processings, setProcessings] = useState<number[]>([]);
+
+  const filterTodos = () => {
```

Comment:
Consider using a useCallback for filterTodos function to avoid unnecessary re-renders.